daft/catalog/__init__.py

```diff
@@ -216,3 +248,28 @@ def __len__(self) -> int:
 
     def __repr__(self) -> str:
         return f"Identifier('{self._identifier.__repr__()}')"
+
+
+class Table(ABC):
+    """Interface for python table implementations."""
+
+    @property
+    def inner(self) -> object | None:
+        """Returns the inner table object if this is an adapter."""
+
+    # TODO deprecated catalog APIs #3819
+    def to_dataframe(self) -> DataFrame:
+        """DEPRECATED: Please use `read()`."""
+        return self.read()
+
+    @abstractmethod
+    def read(self) -> DataFrame:
+        """Returns a DataFrame from this table."""
```

Comment:
Is this valid syntax? 😮 

Does it not have to be followed by a `...` or something

Reply:
hah I didn't even notice! I would think since the body is non-empty? I think the doc comment is an unassigned mutli-line string